app/pages/RemoveNirmsPage.scala

```diff
@@ -16,11 +16,24 @@
 
 package pages
 
+import models.UserAnswers
 import play.api.libs.json.JsPath
 
+import scala.util.Try
+
 case object RemoveNirmsPage extends QuestionPage[Boolean] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "removeNirms"
+
+  override def cleanup(
+    value: Option[Boolean],
+    updatedUserAnswers: UserAnswers,
+    originalUserAnswers: UserAnswers
+  ): Try[UserAnswers] = updatedUserAnswers.get(NirmsNumberUpdatePage) match {
+    case Some(_) if value.isDefined => updatedUserAnswers.set(HasNirmsUpdatePage, !value.get)
```

Comment:
no sure of the use of get here - can you make it more scala functional and do a map or match or something?

Reply:
could do with a test also.

Interested to see how you could make it more functional. trying a few things at the moment but not getting far!

Reply:
Ok I do have something that might work. Let me know if you need it :)

Reply:
Done :)

Reply:
Snap! I got to the same result refactoring

Reply:
Haha!! Nice